core-default/src/main/java/me/aurium/beetle/defaults/task/CommonTaskRunner.java

```diff
@@ -0,0 +1,48 @@
+package me.aurium.beetle.defaults.task;
+
+import me.aurium.beetle.api.task.SyncQueue;
+import me.aurium.beetle.api.task.TaskRunner;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+import java.util.function.Supplier;
+
+public class CommonTaskRunner implements TaskRunner {
+
+    private final Executor asyncQueue;
+    private final Executor syncQueue;
+
+    public CommonTaskRunner(Executor syncQueue, Executor asyncQueue) {
+        this.asyncQueue = asyncQueue;
+        this.syncQueue = syncQueue;
+    }
+
+    @Override
+    public void executeSync(Runnable runnable) {
+        supplySync(() -> {
```

Comment:
Beware, by using supplySync which returns a future, you have just swallowed any exceptions which the runnable throws. Any thrown exceptions are stored in the returned future

Reply:
resolved, i think